packages/shared/src/hooks/useProfileForm.ts

```diff
@@ -89,15 +90,16 @@ const useProfileForm = ({
     ResponseError,
     UpdateProfileParameters
   >(
-    ({ image, ...data }) =>
+    ({ image, onUpdateSuccess, ...data }) =>
       request(graphqlUrl, UPDATE_USER_PROFILE_MUTATION, {
         data,
         upload: image,
       }),
     {
-      onSuccess: async (_, { image, ...vars }) => {
+      onSuccess: async (_, { image, onUpdateSuccess, ...vars }) => {
         setHint({});
         await updateUser({ ...user, ...vars });
+        onUpdateSuccess?.();
         onSuccess?.();
```

Comment:
Hard to understand where these 2 differ

Reply:
It's mainly bad design on the `onSuccess` but didn't want to change ALL use-cases so just added a new one that can be optionally passed to the mutation directly.

If can change it to something as this if you prefer?:
```
if(onUpdateSuccess) {
onUpdateSucces();
} else {
onSuccess?.();
}
```